e2e_tests/tests/test_system.py

```diff
@@ -526,3 +526,18 @@ def test_perform_initial_validation() -> None:
         config, conf.fixtures_path("no_op"), 1, has_zeroth_step=True
     )
     exp.assert_performed_initial_validation(exp_id)
+
+
+@pytest.mark.parallel  # type: ignore
+def test_distributed_logging() -> None:
+    config = conf.load_config(conf.fixtures_path("pytorch-no-op/const.yaml"))
+    config = conf.set_slots_per_trial(config, 8)
+    config = conf.set_max_length(config, {"batches": 1})
+
+    e_id = exp.run_basic_test_with_temp_config(config, conf.fixtures_path("pytorch-no-op"), 1)
+    t_id = exp.experiment_trials(e_id)[0]["id"]
+
+    for i in range(config["resources"]["slots_per_trial"]):
+        assert exp.check_if_string_present_in_trial_logs(
+            t_id, "finished train_batch for rank {}".format(i)
```

Comment:
blocking: Hmm, does this test actually work? I'm only seeing the log lines from rank 0 in my manual testing, presumably because of [this bit](https://github.com/determined-ai/determined/blob/bd06f35d41561137b2d8a28dc9ad653f3c401ab2/harness/determined/_trial_controller.py#L270) that sets the log level to WARNING for non-chief processes (unless debug mode is on, which I don't think is the case here).

Reply:
Yeah, I only ran this test in the _non_ distributed case (commenting out the slots per trial line), so I'll need to fix that. You'll have to let me know how you tested this locally later (I don't think just artificial slots works).

Reply:
You can also run with `debug: true` or use `print()`. Also not sure if the logs get jumbled or not but you might have to do something like:

```
time.sleep(self.context.distrubuted.get_rank())
logging.info(...)
time.sleep(self.context.distributed.get_size() - self.context.distrubuted.get_rank())
```

Reply:
I vote for `print()`, since debug levels should not affect if this test works or not.

Reply:
fixed, all tests are green now, too.

Reply:
I tested this by spinning up two GCP instances and building and running things there. (One might also set up a single 2-GPU instance and run an agent locked to each GPU, but at the time I vaguely recalled that making NCCL or something sad somehow. Though, now that I think about it, that might actually have been about running two agents on the _same_ GPU.)

Reply:
I'm pretty sure it used to not work because we only supported dtrain with host networking mode, and we also hardcoded ports everywhere (we still have quite a few hardcoded ports laying around).

Reply:
> I tested this by spinning up two GCP instances and building and running things there.

Ah, ok. I thought for a second you had tested it locally somehow (not in the cloud) and I was curious.